src/app/pages/my-request-detail/my-request-detail.component.ts

```diff
@@ -207,21 +238,135 @@ export class MyRequestDetailComponent implements OnInit {
       });
   }
 
-  removeItem(event: DataElementDeleteEvent) {
-    const item = event.item;
-    this.removeOneOrMoreDataElements([item]);
+  removeItem(event: DataItemDeleteEvent) {
+    if (!event.dataSchema) {
+      this.toastr.error('Data schema undefined', 'Unable to delete items');
+    } else {
```

Comment:
I would suggest less indentation here, my personal code style tends to try and avoid if-elses and return as soon as possible, for instance:

```ts
if (!event.dataSchema) {
  this.toastr.error();
  return;
}

if (!userFacingText) {
  this.toastr.error();
  return;
}

// ... and so on
```

Reply:
Updated as suggested